src/ok.rs

```diff
@@ -605,3 +671,209 @@ fn memlock_rlimit() -> TestResult {
         mesg: Some(mesg),
     }
 }
+
+fn msr_bit_read(reg:u32, cpu:u16) -> Result<u64,anyhow::Error> {
+    let mut msr = Msr::new(reg, cpu).context("Error Reading MSR")?;
+    let raw_value = msr.read()?;
+    Ok(raw_value)
+}
+
+fn sme_test() -> TestResult{
+        let raw_value = match msr_bit_read(0xC0010010,0) {
+            Ok(raw) => raw,
+            Err(e) => return TestResult{
+                        name: "SME Test".to_string(),
+                        stat: TestState::Fail,
+                        mesg: format!("MSR read failed: {}", e).into()
+            }
+        };
+        let (testres,mesg) = match raw_value >> 23 & 1{
+            1 => (TestState::Pass, "Enabled"),
+            0 => (TestState::Fail, "Disabled"),
+            _ => (TestState::Fail, "Invalid Number Found")
+        };
+        TestResult { 
+            name: "SME enabled".to_string(), 
+            stat: testres, 
+            mesg: Some(mesg.to_string())
+        }
+    }
+
+fn snp_test() -> TestResult {
+    let raw_value = match msr_bit_read(0xC0010010,0) {
+        Ok(raw) => raw,
+        Err(e) => return TestResult{
+                    name: "Secure Nested Paging (SNP)".to_string(),
```

Comment:
Since you already have a test that prints out `Secure Nested Paging (SNP)`, maybe relabel this one as `SNP Enabled`.